src/templates/finops-hub/modules/infrastructure.bicep

```diff
@@ -213,7 +213,7 @@ resource blobPrivateDnsZone 'Microsoft.Network/privateDnsZones@2024-06-01' = if
     properties: {
       registrationEnabled: false
       virtualNetwork: {
-        id: hub.routing.networkId
+        id: vNet.id
```

Comment:
FYI - I was on the fence whether to use an on-demand reference or the ID that we have to maintain anyway. I ran into issues using the reference in some cases so I tended to use the original ID to work around bicep dependency limitations. I'm not overly opinionated, however. Just wanted to share the reasoning behind it being an ID rather than a reference.